molo/core/content_import/helpers/importing.py

```diff
@@ -0,0 +1,229 @@
+import json
+
+from babel import Locale
+
+from molo.core.models import (
+    Main, SiteLanguage, PageTranslation, SectionPage, ArticlePage, FooterPage)
+from molo.core.content_import.helpers.get_image import get_image_file
+
+from unicore.content.models import Category, Page
+
+
+def import_repo(repo, main_locale, children, should_nest=False):
+    main = Main.objects.all().first()
+
+    for locale in [main_locale] + children:
+        lang = create_language(repo, locale, locale == main_locale)
+        parent = get_or_create_main_section(repo, lang, main, should_nest)
+        import_locale_content(repo, lang, parent)
+
+    update_pages_with_linked_page_field(repo)
+
+
+def get_models(repo, cls, **kw):
+    qs = repo.workspace.S(cls)
+    qs = qs.filter(**kw)
+
+    # S() only returns 10 results if you don't ask for more
+    return qs.order_by('position')[:10000]
+
+
+def get_or_create_main_section(repo, lang, main, should_nest):
+    if should_nest:
+        return main_section_for_language(repo, lang, main)
+    else:
+        return main
+
+
+def create_language(repo, locale, is_main):
+    language, _ = SiteLanguage.objects.get_or_create(
+        locale=Locale.parse(locale).language,
+        is_main_language=is_main)
+
+    return {
+        'locale': locale,
+        'language': language,
+    }
+
+
+def import_locale_content(repo, lang, parent):
+    import_all_categories(repo, lang, parent)
+    import_all_pages(repo, lang)
+
+
+def main_section_datum(repo, lang):
+    # NOTE For sections that correspond to a repository (for the multi-repo
+    # import case), We can't simply rely on the ORM to create a uuid for us,
+    # since we need to retrieve the section if it already exists instead of
+    # creating it again. This means we need to find the section by its
+    # corresponding repository's name and the locale that the section was
+    # created for.  At the moment, we use the uuid field for this, though isn't
+    # quite right -- uuid's need to be universally unique. A more robust
+    # solution might be to support a metadata field on the SectionPage model
+    # and query by this field for this case
+
+    # NOTE At the moment, we are using the repository title as the title for
+    # the corresponding section we create, regardless of the language the
+    # section is created for. We need a way of creating a title from the
+    # repository in the given language
```

Comment:
The challenge here is that the translation for the repository title lives in the `.po` file of the `springboard-iogt` codebase, and not available in the repositories. I think this is an exception (or shortcoming) and we should explain it to the PM's/Client.

They'll have to manually updated the translation after the import. Alternatively, we can create a mapping dict that stores the translations that we know of somewhere...but this is less clean..